examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,409 @@
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Developer makes use of Sangoma NetBorder Gateway or Sangoma Session
+#     Border Controller
+#  2. Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#  3. Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+
+import time
+import switchy
+import threading
+from switchy.marks import event_callback
+from switchy import get_originator
+
+# Enable logging to stderr
+# Debug levels: 'INFO' for production, 'DEBUG' for development
+log = switchy.utils.log_to_stderr('INFO')
+
+# Specify FreeSWITCH or Sangoma NSG IP information
+# In this example the sample app is running on
+# Sangoma NSG appliance hence the use of local address
+host = "127.0.0.1"
+port = 8821
+
+
+class IVRCallLogic(object):
+    """
+    IVR Call Logic switchy app.
+    All custom development should be done on this class.
+
+    You can use the following variable types:
+        self.<var_name> instance variables are global in nature.
+        call.vars.['<var_name>'] should be used for per call info and state.
+    """
+    def prepost(self, client, listener):
+        """Defines a fixture-like pre/post app load hook for performing
+        provisioning steps before this app is loaded.
+        """
+        # Get the install directory of FreeSWITCH or Sangoma NSG and append
+        # recording to it. By default, on Sangoma systems, FreeSWITCH resides
+        # in /usr/local/freeswitch, NSG resides in /usr/local/nsg.
+        self.base_dir = client.cmd('global_getvar base_dir')
+        self.recdir = self.base_dir + "/recording"
+        log.info("Setting recording dir to '{}".format(self.recdir))
+
+        # Get the install directory of NSG and append sounds to it
+        self.sound_dir = "{}/{}".format(
+            self.base_dir, 'sounds/en/us/callie/ivr/8000')
```

Comment:
Yeah only issue is that the FS isn't necessarily that of the OS switchy is running on.
Totally agree though.